benchmarks/inference-server/triton/manifest-templates/triton-tensorrtllm-inference-gs.tftpl

```diff
@@ -0,0 +1,76 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: triton-infer-gcs
+  namespace: ${namespace}  # Replace ${namespace} with your actual namespace
+  labels:
+    app: triton-infer-gcs
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: triton-infer-gcs
+  template:
+    metadata:
+      labels:
+        app: triton-infer-gcs
+    spec:
+      serviceAccountName: ${ksa}  # Replace ${ksa} with your Kubernetes Service Account name
+      volumes:
+        - name: dshm
+          emptyDir:
+            medium: Memory
+            sizeLimit: 1Gi
+        - name: all-models-volume
+          emptyDir: {}
+      initContainers:
+        - name: init-gcs-download
+          image: google/cloud-sdk
+          serviceAccountName: ${ksa}
+          command: ["/bin/sh", "-c"] 
+          args:
+            - gsutil cp -r gs://your_gcs_repo/all_models ./;
+          volumeMounts:
+            - name: all-models-volume
+              mountPath: /all_models
+      containers:
+        - name: triton-inference
+          image: "${image_path}"  # Replace ${image_path} with the actual image path
+          workingDir: /opt/tritonserver
+          #command: ["/bin/sleep", "3600"]
+          command: ["/bin/bash", "-c"]
+          args: ["pip install sentencepiece protobuf && huggingface-cli login --token $HUGGINGFACE_TOKEN && mpirun --allow-run-as-root -n 1 /opt/tritonserver/bin/tritonserver --model-repository=/all_models/inflight_batcher_llm --disable-auto-complete-config --backend-config=python,shm-region-prefix-name=prefix0_ :"]
```

Comment:
Ah I missed this earlier - this still utilizes custom image path.

My understanding with tensorrtllm (based on their documentation, not from testing it out myself yet) is that they have a base image that we can utilize as long as we provide the prebuilt model weights - ref: https://github.com/triton-inference-server/tensorrtllm_backend/blob/main/README.md#run-the-pre-built-docker-container

This -gs.yaml is for the pathway for customers to pull in models that are stored in external storage (thus not built into the image), and it should rely on the base triton pathway, not the custom one. I'm assuming if we use the base image these packages are already pre-installed.

Reply:
In this implementation, we utilize the base image nvcr.io/nvidia/tritonserver:23.10-trtllm-python-py3. However, we've made this a variable to allow users the flexibility to upgrade to newer versions as they become available. It may be beneficial to elaborate on this option further in the Readme for clarity.

Contrary to 'Option 1', outlined in the [Triton Inference Server documentation for TensorRT LL Model backend](https://github.com/triton-inference-server/tensorrtllm_backend/blob/main/README.md#option-1-launch-triton-server-within-triton-ngc-container), Option2 employs a base triton server image provided by Nvidia and specifies all configurations directly in the YAML file, eliminating the need to create a new image. We can intoduce  'Option 3' (or update 'Option 2'), which mirrors 'Option 1' but differs in that the model is loaded via Google Storage.

As of version 23.10, the SentencePiece and protobuf appears to be missing, and a Hugging Face login for llama is required. This is because the tokenizer is downloaded at runtime.

Reply:
Makes sense. If the plan is to make this yaml simpler, then defaulting the image_path to the Nvidia image by default and specifying the custom image as an additional option instead of providing them as two different options that the user has to choose between would be better.

Reply:
Yes, I made the gcs method as default and custoem image as an additional option